hw_12/index.php

```diff
@@ -0,0 +1,258 @@
+<?php
+
+//declare(strict_types=1);
+error_reporting(E_ALL);
+ini_set('display_errors', '1');
+
+/*///////////NORMAL LEVEL////////////*/
+
+/*
+//Task 1
+function message()
+{
+    echo '«Hello, World!»';
+}
+
+message();
+*/
+
+/*
+//Task 2
+$year = 2022;
+function leapYear(): bool
+{
+    global $year;
+    if ($year % 4 == 0) {
+        return true;
+    } else {
+        return false;
+    }
+}
+
+var_dump(leapYear());
+*/
+
+/*
+//Task 3
+$str = 'Cascading Style Sheets';
+function abbrev()
+{
+    global $str;
+    $abbrevStr = implode('', array_diff_assoc(str_split(ucwords($str)), str_split(strtolower($str))));
+    echo $abbrevStr;
+}
+
+abbrev();
+*/
+
+/*
+//Task 4
+$strStart = 'Nastya Настя';
+function strReversed()
+{
+    global $strStart;
+    $strFinish = '';
+    // mb_strlen - получение длины строки; mb_substr - возвращает часть строки
+    for ($i = mb_strlen($strStart); $i >= 0; $i--) {
+        $strFinish .= mb_substr($strStart, $i, 1);// .= - объединение строк с присвоением результата
+    }
+    echo $strFinish;
+}
+
+strReversed();
+*/
+
+/*
+//Task 5
+function IsPrime($a): bool
+{
+    for ($i = 2; $i < $a; $i++) {
+        if ($a % $i == 0) {
+            return false;
+        } else {
+            return true;
+        }
+    }
+
+    if ($a == 0) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+var_dump(isPrime(6));
+*/
+
+/*
+//Task 6
+function wordLenFinish($str)
+{
+    $arrNewWord = explode(" ", $str,); //explode — Разбивает строку с помощью разделителя
+    $strFinish = rtrim(end($arrNewWord), '!@#$%^&*()_+"№;%:?=-');
+//rtrim-убирает пробелы и заданные символы справа  // end-выводит последний элемент массива
+    echo iconv_strlen($strFinish); //iconv_strlen — Возвращает количество символов в строке
+}
+
+wordLenFinish('задание №6 PHP!');
+*/
+
+/*
+//Task 7
+function maxNumArr($arr)
+{
+    if (1 === count($arr)) {
+        return $arr;
+    }
+    $result = array();
+    foreach ($arr as $key => $item) {
+        foreach (maxNumArr(array_diff_key($arr, array($key => $item))) as $newArr)
+            //array_diff_key — Вычисляет расхождение массивов, сравнивая ключи
+        {
+            $result[] = $item . $newArr;
+        }
+    }
+    return $result;
+}
+
+$arr = [3, 24, 4];
+echo max(maxNumArr($arr));
+*/
+
+/*
+//Task 8
+function anagramma($word1, $word2): bool
+{
+    $arr1 = str_split($word1);
+    $arr2 = str_split($word2);
+    $w1 = strlen($word1);
+    $w2 = strlen($word2);
+
+    if ((!array_diff($arr1, $arr2)) and ($w1 == $w2)) {
+        return true;
+    } else {
+        return false;
+    }
```

Comment:
if лишний, and не используем, используем &&
```suggestion
   return !array_diff($arr1, $arr2)) && ($w1 == $w2);
```